src/components/ParamsBox.tsx

```diff
@@ -69,13 +74,16 @@ const ParamsBox: FunctionComponent<{}> = () => {
 						</select>
 					)}
 					{FiniteParams.is(selectedEntity.params) && (
-						<input type="text" pattern={pattern} value={selectedEntity.params.stringFormat} onChange={onChangeMagazine}/>
+						<input type="text" pattern="^(?:b|g)+$" value={selectedEntity.params.stringFormat} onChange={onChangeMagazine}/>
 					)}
 				</Fragment>
 			)}
-			{!Hoppi.is(selectedEntity) &&  (
+			{Text.is(selectedEntity) &&  (
+				<input type="text" value={Text} onChange={onChangeText}/>
+			)};
+			{!Hoppi.is(selectedEntity) && !Text.is(selectedEntity) (
```

Comment:
```suggestion
			{!Hoppi.is(selectedEntity) && !Text.is(selectedEntity) && (
```

You forgot the final and. :slightly_smiling_face: